runtime/tests/ecdsa.rs

```diff
@@ -12,140 +11,138 @@ use zerocopy::AsBytes;
-                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-                0x00, 0x00, 0x00, 0x00, 0x00, 0x01,
-            ],
-        },
-    ];
-
-    for c in cases {
-        // Stream to SHA ACC
-        model
-            .compute_sha512_acc_digest(c.msg, ShaAccMode::Sha384Stream)
+    #[allow(dead_code)]
+    #[derive(Debug)]
+    struct WycheproofResults {
+        id: usize,
+        comment: String,
+    }
+    // Collect all the errors and print at the end
+    let mut wyche_err: Vec<WycheproofResults> = Vec::new();
+    let mut wyche_ran: Vec<WycheproofResults> = Vec::new();
+    let test_set =
+        wycheproof::ecdsa::TestSet::load(wycheproof::ecdsa::TestName::EcdsaSecp384r1Sha384)
             .unwrap();
+    // The mailbox has a fixed len for x and y so filter out untestable cases
+    let test_groups = test_set
+        .test_groups
+        .iter()
+        .filter(|x| x.key.affine_x.as_slice().len() <= 48)
+        .filter(|x| x.key.affine_y.as_slice().len() <= 48);
+    for test_group in test_groups {
+        for test in &test_group.tests {
+            // Since the mailbox uses R, S as input for signature use only tests with a valid signature
+            let Ok(signature) = openssl::ecdsa::EcdsaSig::from_der(test.sig.as_bytes()) else {
+                continue;
+            };
+            if signature.r().to_vec().len() > 48 || signature.s().to_vec().len() > 48 {
+                continue;
+            }
+            // openssl rust crates has problems parsing some DER signatures. Skip those as those can't be
+            // sent to the mailbox anyway.
+            if [
+                "Legacy: ASN encoding of s misses leading 0",
+                "length of sequence [r, s] uses long form encoding",
+                "length of sequence [r, s] contains a leading 0",
+                "appending unused 0's to sequence [r, s]",
+                "indefinite length",
+                "indefinite length with garbage",
+                "length of r uses long form encoding",
+                "length of r contains a leading 0",
+                "prepending 0's to r",
+                "length of s uses long form encoding",
+                "length of s contains a leading 0",
+                "prepending 0's to s",
+            ]
+            .contains(&test.comment.as_str())
+            {
+                continue;
+            }
 
-        let mut cmd = EcdsaVerifyCmd {
-            chksum: 0,
-            pub_key_x: c.pub_x,
-            pub_key_y: c.pub_y,
-            signature_r: c.sig_r,
-            signature_s: c.sig_s,
-        };
+            fn report_test(err: &mut Vec<WycheproofResults>, id: usize, comment: &str) {
+                err.push(WycheproofResults {
+                    id,
+                    comment: comment.to_owned(),
+                });
+            }
```

Comment:
This functions replaces one line of code with one line of code. Can probably just call this inline where needed.